File: src/main/java/de/gesellix/docker/client/filesocket/FileSocket.java
package de.gesellix.docker.client.filesocket;

import java.net.InetAddress;
import java.net.Socket;

public abstract class FileSocket extends Socket {

  public final static String SOCKET_MARKER = ".socket";

  public String encodeHostname(String hostname) {
    return new HostnameEncoder().encode(hostname) + SOCKET_MARKER;
  }

  public String decodeHostname(InetAddress address) {
    String hostName = address.getHostName();
    if (!hostName.endsWith(SOCKET_MARKER)) {
      return hostName;
    }
    return new HostnameEncoder().decode(hostName.substring(0, hostName.indexOf(SOCKET_MARKER)));
  }
}

File: src/main/kotlin/de/gesellix/docker/client/filesocket/FileSocket.kt
package de.gesellix.docker.client.filesocket

import java.net.InetAddress
import java.net.Socket

abstract class FileSocket : Socket() {

    companion object {
        const val SOCKET_MARKER: String = ".socket"
    }

    fun encodeHostname(hostname: String): String {
        return "${HostnameEncoder().encode(hostname)}${SOCKET_MARKER}"
    }

    fun decodeHostname(address: InetAddress): String {
        val hostName = address.hostName
        return HostnameEncoder().decode(hostName.substring(0, hostName.indexOf(SOCKET_MARKER)))
    }
}
